src/core/lib/iomgr/call_combiner.h

```diff
@@ -109,4 +110,79 @@ void grpc_call_combiner_set_notify_on_cancel(grpc_call_combiner* call_combiner,
 void grpc_call_combiner_cancel(grpc_call_combiner* call_combiner,
                                grpc_error* error);
 
+namespace grpc_core {
+
+// Helper for running a list of closures in a call combiner.
+//
+// Each callback running in the call combiner will eventually be
+// returned to the surface, at which point the surface will yield the
+// call combiner.  So when we are running in the call combiner and have
+// more than one callback to return to the surface, we need to re-enter
+// the call combiner for all but one of those callbacks.
+class CallCombinerClosureList {
+ public:
+  CallCombinerClosureList() {}
+
+  // Adds a closure to the list.  The closure must eventually result in
+  // the call combiner being yielded.
+  void Add(grpc_closure* closure, grpc_error* error, const char* reason) {
+    closures_.emplace_back(closure, error, reason);
+  }
+
+  // Runs all closures in the call combiner.
+  //
+  // If yield_call_combiner is true, then the call combiner will be yielded.
+  // In this case, all but one of the closures will be scheduled via
+  // CALL_COMBINER_START(), and the remaining closure will be scheduled
+  // directly via GRPC_CLOSURE_SCHED(), which will eventually result in
+  // yielding the call combiner.  If the list is empty, then the call
+  // combiner will be yielded immediately.
+  //
+  // If yield_call_combiner is false, then the call combiner will not be
+  // yielded.  All callbacks will be scheduled via GRPC_CALL_COMBINER_START().
+  void RunClosures(grpc_call_combiner* call_combiner,
```

Comment:
Style nit; can you split this into two public APIs? `RunClosures ` and then `RunClosuresAndYieldCallCombiner` (or something similar).

Then you wouldn't have to worry about calling this like
```
closures.RunClosures(calld->call_combiner, true /* yield_call_combiner */);
```